<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GUIClass</class>
 <widget class="QMainWindow" name="GUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>748</width>
    <height>531</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>748</width>
    <height>531</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Simple image editor</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/GUI/Resources/mainicon.png</normaloff>:/GUI/Resources/mainicon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="gbImage">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>181</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QPushButton" name="btnOpen">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>161</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSave">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>161</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSaveAs">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>161</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Save As...</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbFilters">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>180</width>
      <height>341</height>
     </rect>
    </property>
    <property name="title">
     <string>Filters</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
    <widget class="QPushButton" name="btnGreyWorld">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>95</y>
       <width>160</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Grey World</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnMedian">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>115</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Median</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGrayscale">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>160</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Grayscale</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnHygramLinear">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>45</y>
       <width>160</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Autoconrast</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnHygramChennel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>160</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Autolevels</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="editMedian">
     <property name="geometry">
      <rect>
       <x>125</x>
       <y>120</y>
       <width>35</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Matrix radious</string>
     </property>
     <property name="statusTip">
      <string>Set matrix size</string>
     </property>
     <property name="whatsThis">
      <string>Set matrix size</string>
     </property>
     <property name="accessibleName">
      <string/>
     </property>
     <property name="accessibleDescription">
      <string/>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>6</string>
     </property>
     <property name="maxLength">
      <number>3</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="labelWarning">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>320</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 0, 0);</string>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGaussian">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>145</y>
       <width>70</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Gaussian</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="editGaussianSigma">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>145</y>
       <width>45</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Sigma</string>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string>1</string>
     </property>
     <property name="maxLength">
      <number>4</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="editGaussian">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>125</x>
       <y>145</y>
       <width>35</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Matrix radious</string>
     </property>
     <property name="text">
      <string>6</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSobel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Sobel</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnScale">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>260</y>
       <width>70</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Scale</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="editScale">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>260</y>
       <width>70</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Scale Factor</string>
     </property>
     <property name="statusTip">
      <string>Scale factor</string>
     </property>
     <property name="maxLength">
      <number>9</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="px_label0">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>125</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>px</string>
     </property>
    </widget>
    <widget class="QLabel" name="px_label1">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>150</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>px</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>265</y>
       <width>25</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>tms</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnRotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>235</y>
       <width>125</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="editRotate">
     <property name="geometry">
      <rect>
       <x>135</x>
       <y>235</y>
       <width>35</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Rotation angle</string>
     </property>
     <property name="statusTip">
      <string>Rotation angle</string>
     </property>
     <property name="text">
      <string>90</string>
     </property>
     <property name="maxLength">
      <number>4</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>167</x>
       <y>235</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>°</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnManually">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>300</y>
       <width>85</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Custom</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="editManuallyX">
     <property name="geometry">
      <rect>
       <x>95</x>
       <y>300</y>
       <width>35</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Matrix width</string>
     </property>
     <property name="text">
      <string>3</string>
     </property>
     <property name="maxLength">
      <number>3</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="editManuallyY">
     <property name="geometry">
      <rect>
       <x>135</x>
       <y>300</y>
       <width>35</width>
       <height>23</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Matrix height</string>
     </property>
     <property name="text">
      <string>3</string>
     </property>
     <property name="maxLength">
      <number>3</number>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>305</y>
       <width>10</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>x</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnGlass">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>195</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Glass</string>
     </property>
    </widget>
    <widget class="Line" name="line">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>220</y>
       <width>180</width>
       <height>12</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="Line" name="line_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>285</y>
       <width>180</width>
       <height>12</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QGraphicsView" name="gvMain">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>10</y>
      <width>541</width>
      <height>471</height>
     </rect>
    </property>
    <property name="cursor" stdset="0">
     <cursorShape>CrossCursor</cursorShape>
    </property>
   </widget>
   <widget class="QProgressBar" name="pbMain">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>490</y>
      <width>731</width>
      <height>16</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="textVisible">
     <bool>true</bool>
    </property>
    <property name="invertedAppearance">
     <bool>false</bool>
    </property>
    <property name="format">
     <string>%p%</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>748</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Filters</string>
    </property>
    <addaction name="mnuGrayscale"/>
    <addaction name="mnuHygramLinear"/>
    <addaction name="mnuHygramChennel"/>
    <addaction name="mnuGreyWorld"/>
    <addaction name="mnuMedian"/>
    <addaction name="mnuGaussian"/>
    <addaction name="mnuSobel"/>
    <addaction name="mnuScale"/>
    <addaction name="mnuRotate"/>
    <addaction name="mnuManually"/>
    <addaction name="mnuGlass"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="mnuAbout"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="mnuOpen"/>
    <addaction name="mnuSave"/>
    <addaction name="mnuSaveAs"/>
    <addaction name="separator"/>
    <addaction name="mnuExit"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
  </widget>
  <action name="mnuOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="mnuSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="mnuSaveAs">
   <property name="text">
    <string>Save As ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_4">
   <property name="text">
    <string>-</string>
   </property>
  </action>
  <action name="mnuExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="action_7">
   <property name="text">
    <string>&quot;Серый мир&quot;</string>
   </property>
  </action>
  <action name="action_8">
   <property name="text">
    <string>Медианная фильтрация</string>
   </property>
  </action>
  <action name="mnuAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="mnuGrayscale">
   <property name="text">
    <string>Grayscale</string>
   </property>
  </action>
  <action name="mnuHygramLinear">
   <property name="text">
    <string>Hystogram Linear</string>
   </property>
  </action>
  <action name="mnuHygramChennel">
   <property name="text">
    <string>Per-Chennel hystogram</string>
   </property>
  </action>
  <action name="mnuGreyWorld">
   <property name="text">
    <string>Grey World</string>
   </property>
  </action>
  <action name="mnuMedian">
   <property name="text">
    <string>Median</string>
   </property>
  </action>
  <action name="mnuGaussian">
   <property name="text">
    <string>Gaussain</string>
   </property>
  </action>
  <action name="mnuSobel">
   <property name="text">
    <string>Sobel</string>
   </property>
  </action>
  <action name="mnuScale">
   <property name="text">
    <string>Scaling</string>
   </property>
  </action>
  <action name="mnuRotate">
   <property name="text">
    <string>Rotate</string>
   </property>
  </action>
  <action name="mnuManually">
   <property name="text">
    <string>Custom</string>
   </property>
  </action>
  <action name="mnuGlass">
   <property name="text">
    <string>Glass</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>btnOpen</tabstop>
  <tabstop>btnSave</tabstop>
  <tabstop>btnSaveAs</tabstop>
  <tabstop>btnGrayscale</tabstop>
  <tabstop>btnHygramLinear</tabstop>
  <tabstop>btnHygramChennel</tabstop>
  <tabstop>btnGreyWorld</tabstop>
  <tabstop>btnMedian</tabstop>
  <tabstop>editMedian</tabstop>
  <tabstop>btnGaussian</tabstop>
  <tabstop>editGaussianSigma</tabstop>
  <tabstop>editGaussian</tabstop>
  <tabstop>gvMain</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>editGaussianSigma</sender>
   <signal>textEdited(QString)</signal>
   <receiver>GUIClass</receiver>
   <slot>onEditGaussianSigma_TextChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>462</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editGaussian</sender>
   <signal>textEdited(QString)</signal>
   <receiver>GUIClass</receiver>
   <slot>onEditGaussian_TextChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>306</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>437</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnGaussian</sender>
   <signal>clicked()</signal>
   <receiver>GUIClass</receiver>
   <slot>onGaussian()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>293</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>401</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editScale</sender>
   <signal>returnPressed()</signal>
   <receiver>btnScale</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>159</x>
     <y>399</y>
    </hint>
    <hint type="destinationlabel">
     <x>77</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editGaussian</sender>
   <signal>returnPressed()</signal>
   <receiver>btnGaussian</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>167</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editRotate</sender>
   <signal>returnPressed()</signal>
   <receiver>btnRotate</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>389</y>
    </hint>
    <hint type="destinationlabel">
     <x>131</x>
     <y>389</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onGaussian()</slot>
 </slots>
</ui>
